bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -4575,6 +4582,13 @@ ReplicaImp::~ReplicaImp() {
 
 void ReplicaImp::stop() {
   LOG_DEBUG(GL, "ReplicaImp::stop started");
+
+  // stop components dependent on ReplicaImp
+  stopCallbacks_.invokeAll();
+
+  internalThreadPool.stop();
+  postExecThread_.stop();
+
```

Comment:
Do we have two mechanisms now to stop external objects? One from InternalReplica and another one here?